modules/Bio/EnsEMBL/Variation/Pipeline/PhenotypeAnnotation/ImportGENCC.pm

```diff
@@ -0,0 +1,259 @@
+package Bio::EnsEMBL::Variation::Pipeline::PhenotypeAnnotation::ImportGENCC;
+
+use strict;
+use warnings;
+use base qw/Bio::EnsEMBL::Variation::Pipeline::PhenotypeAnnotation::BasePhenotypeAnnotation/;
+
+use Text::CSV_XS;
+use POSIX qw(strftime);
+use File::Path qw(make_path);
+use HTTP::Tiny;
+use Bio::EnsEMBL::Registry;
+
+sub fetch_input {
+  my ($self) = @_;
+
+  my $species = $self->param_required('species');
+  my $pdir    = $self->param_required('pipeline_dir');
+
+  my $workdir = $pdir . "/GenCC/" . $species;
+  unless (-d $workdir) {
+    my $err;
+    make_path($workdir, { error => \$err });
+    $self->throw("make_path failed to create $workdir") if $err && @$err;
+  }
+  $self->workdir($workdir);
+
+  open(my $logFH,     ">", $workdir."/log_import_out_GenCC_".$species)
+    || $self->throw("Failed to open log_import_out_GenCC_$species: $!");
+  open(my $errFH,     ">", $workdir."/log_import_err_GenCC_".$species)
+    || $self->throw("Failed to open log_import_err_GenCC_$species: $!");
+  open(my $pipelogFH, ">", $workdir."/log_import_debug_pipe_GenCC_".$species)
+    || $self->throw("Failed to open log_import_debug_pipe_GenCC_$species: $!");
+  $self->logFH($logFH);
+  $self->errFH($errFH);
+  $self->pipelogFH($pipelogFH);
+
+  my $default_dir  = $pdir . "/GenCC";
+  my $default_file = $default_dir . "/gencc.csv";
+  my $file         = $self->param('gencc_file') // $default_file;
+
+  # If file is missing/empty, fetch the latest CSV from GenCC
+  unless (-s $file) {
+    make_path($default_dir);
+    my $url  = 'https://search.thegencc.org/download/action/submissions-export-csv';
+    my $http = HTTP::Tiny->new(timeout => 180);
+    my $res  = $http->get($url);
+    $self->throw("GenCC download failed: $res->{status} $res->{reason}") unless $res->{success};
+    open(my $OUT, ">:encoding(UTF-8)", $default_file) or $self->throw("Cannot write $default_file: $!");
+    print $OUT $res->{content};
+    close $OUT;
+    $file = $default_file;
+  }
+
+  -s $file or $self->throw("GenCC input CSV not found or empty after download: $file");
+
+  my $version = $self->param('gencc_version') // strftime("%Y%m%d", localtime);
+
+  $self->param('input_file',    $file);
+  $self->param('gencc_version', $version);
+}
+
+sub run {
+  my ($self) = @_;
+
+  my $species = $self->param_required('species');
+  my $file    = $self->param_required('input_file');
+  my $version = $self->param_required('gencc_version');
+
+  my %skip = map { $_ => 1 } @{ $self->param('filter_submitters') || [ 'G2P' ] };
```

Comment:
You could also skip `Orphanet` as it is already imported by this pipeline in a different step.